### PR TITLE
BIP 300: Various improvements

### PR DESCRIPTION
**Pull request by luke-jr**

### CODE REVIEW

File: bip-0300.mediawiki [hunk 1]
```diff
@@ -287,11 +287,15 @@ The upvote vector will code "abstain" as 0xFF (or 0xFFFF); it will code "alarm"
 
 For example: if there are two sidechains, and we wish to upvote the 7th bundle on sidechain #1 plus the 4th bundle on sidechain #2, then the upvote vector would be { 07, 04 }. And M4 would be [0x6A,D77D1776,00,0006,0003].
 
-The version number allows us to shrink the upvote vector in many cases. Version 0x00 requires a full two bytes per sidechain, but it always works. Version 0x01 uses half that (one byte per sidechain), and it works while all sidechains have fewer than 255 disputed withdrawals (ie, 99.99%+ of the time). Version 0x02 uses zero bytes (ie, 6 bytes for the whole M4) and sets this block's M4 equal to the previous block's M4. Version 0x03 upvotes only those withdrawals that are leading their rivals by at least 50 votes.
+The version number allows us to shrink the upvote vector in many cases.
+Version 0x00 omits the upvote vector entirely (ie, 6 bytes for the whole M4) and sets this block's M4 equal to the previous block's M4.
```
Comment (staffserg) [inline bip-0300.mediawiki:291]: Awesome

**Review by staffserg — APPROVED** [review]
Awesome

**Comment by psztorc** [comment]
These changes are good

**Comment by immense055** [comment]
ac80e11298390f24dbf7160a48edec9142ac0085

**Comment by ChrisMartl** [comment]
@psztorc 

Can you please provide a statement how this proposal will affect the Nash equilibrium between the distribution of affordable archival full nodes and affordable mining devices?

Can you please provide a statement how the concern, that equilibrium is reached by highly capitalized and centralized original node (storage + mining) entities will be mitigated?

**Comment by immense055** [comment]
c9098970a5812c12d63e2683b4f815845b508681

**Comment by psztorc** [comment]
> @psztorc 
> 
> Can you please provide a statement how this proposal will affect the Nash equilibrium between the distribution of affordable archival full nodes and affordable mining devices?

Long term, this proposal makes it easier to run a full node, since it allows us to shrink the L1 blocksize. It also makes it easier to mine, since a higher percentage of revenue comes from low-fixed-cost activities.

> Can you please provide a statement how the concern, that equilibrium is reached by highly capitalized and centralized original node (storage + mining) entities will be mitigated?

Node costs will never be significant (in comparison to ASIC costs or physical security) -- for any sidechain, there is a group of people running nodes for free (else the network dies).

If they end up being significant anyway, then miners will utilize blind merged mining (or a custodial version of it) to shirk the cost. If the cost is persistently high anyway, then the L2 network will die off. L1 is unaffected.

**Comment by ChrisMartl** [comment]
@psztorc 

Can you please provide empirical observations from other experimental projects like e.g. Ethereum Classic or Ethereum PoW, where this concept (BIP300) (or equivalent for those projects) has being applied and its short, medium and long term effects on that system as a whole in regards of the concerns being addressed above?

**Comment by psztorc** [comment]
Anyone who is receiving money (on the L2 sidechain), would want to run a sidechain node, for their own peace of mind.

This could include large merchants, large payment processors -- or just really paranoid people. In general, the more users, the more nodes -- which is good from the point of view of Bip301 (which really only requires two, if they are healthy and high-uptime and competing with each other).

**Pull request merged**